server-plugins/board-resources/src/index.ts

```diff
@@ -0,0 +1,56 @@
+//
+// Copyright © 2022 Hardcore Engineering Inc.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import type { Tx, TxCreateDoc, TxRemoveDoc } from '@anticrm/core'
+import type { TriggerControl } from '@anticrm/server-core'
+import type { Card, CardLabel } from '@anticrm/board'
+import board from '@anticrm/board'
+import core, { TxProcessor } from '@anticrm/core'
+
+/**
+ * @public
+ */
+export async function OnLabelDelete (tx: Tx, { findAll, hierarchy, txFactory }: TriggerControl): Promise<Tx[]> {
+  if (tx._class !== core.class.TxRemoveDoc) {
+    return []
+  }
+
+  const rmTx = tx as TxRemoveDoc<CardLabel>
+  if (!hierarchy.isDerived(rmTx.objectClass, board.class.CardLabel)) {
+    return []
+  }
+
+  const createTx = (await findAll(core.class.TxCreateDoc, {
+    objectId: rmTx.objectId
+  }, { limit: 1 }))[0]
+  if (createTx === undefined) {
+    return []
+  }
+
+  const label = TxProcessor.createDoc2Doc(createTx as TxCreateDoc<CardLabel>)
+  const cards = (await findAll<Card>(board.class.Card, { space: label.attachedTo as any }))
```

Comment:
```suggestion
  const cards = (await findAll<Card>(board.class.Card, { space: label.attachedTo as any, labels: label._id }))
```
will find all cards with label inside array, it will be much better.

Reply:
Fixed